File: app/src/androidTest/java/br/com/battista/bgscore/activity/HomeActivityTest.java
package br.com.battista.bgscore.activity;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isAssignableFrom;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static android.support.test.espresso.matcher.ViewMatchers.withParent;
import static android.support.test.espresso.matcher.ViewMatchers.withText;
import static junit.framework.Assert.assertNotNull;
import static org.hamcrest.Matchers.allOf;
import static org.hamcrest.Matchers.equalTo;
import static org.junit.Assert.assertThat;

import android.content.Intent;
import android.support.design.widget.BottomNavigationView;
import android.support.test.filters.LargeTest;
import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;
import android.support.v7.widget.Toolbar;
import android.widget.TextView;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import br.com.battista.bgscore.MainApplication;
import br.com.battista.bgscore.R;
import br.com.battista.bgscore.helper.HomeActivityHelper;
import br.com.battista.bgscore.model.User;

@LargeTest
@RunWith(AndroidJUnit4.class)
public class HomeActivityTest {

    @Rule
    public ActivityTestRule<HomeActivity> testRule =
            new ActivityTestRule<>(HomeActivity.class, Boolean.TRUE, Boolean.FALSE);

    private User user;

    @Before
    public void setup() {
        testRule.launchActivity(new Intent());

        MainApplication application = (MainApplication)
                testRule.getActivity().getApplication();
        MainApplication.init(application);

        user = HomeActivityHelper.createNewUser();
        application.clearPreferences();
        application.setUser(user);
    }

    @Test
    public void checkExistsTheElementsActivity() {
        HomeActivity activity = testRule.getActivity();

        Toolbar toolbar = activity.findViewById(R.id.toolbar);
        assertNotNull(toolbar);

        BottomNavigationView bottomNavigationView =
                activity.findViewById(R.id.bottom_navigation);
        assertNotNull(bottomNavigationView);
    }

    @Test
    public void checkExistMenuInBottomNavigationView() {
        HomeActivity activity = testRule.getActivity();

        BottomNavigationView bottomNavigationView =
                activity.findViewById(R.id.bottom_navigation);
        assertNotNull(bottomNavigationView);

        assertThat(bottomNavigationView.getMenu().size(), equalTo(5));

        assertNotNull(bottomNavigationView.getMenu().findItem(R.id.action_home));
        assertNotNull(bottomNavigationView.getMenu().findItem(R.id.action_matches));
        assertNotNull(bottomNavigationView.getMenu().findItem(R.id.action_games));
        assertNotNull(bottomNavigationView.getMenu().findItem(R.id.action_account));
        assertNotNull(bottomNavigationView.getMenu().findItem(R.id.action_info));
    }

    @Test
    public void shouldShowHomeWhenClickActionHome() {
        try {
            onView(withText(R.string.btn_ok))
                    .perform(click());
        } catch (Exception e) {
        }

        onView(withId(R.id.action_home))
                .perform(click());

        onView(allOf(isAssignableFrom(TextView.class), withParent(isAssignableFrom(Toolbar.class))))
                .check(matches(withText(R.string.title_home)));
    }

    @Test
    public void shouldShowMatchesWhenClickActionMatches() {
        try {
            onView(withText(R.string.btn_ok))
                    .perform(click());
        } catch (Exception e) {
        }

        onView(withId(R.id.action_matches))
                .perform(click());

        onView(allOf(isAssignableFrom(TextView.class), withParent(isAssignableFrom(Toolbar.class))))
                .check(matches(withText(R.string.title_matches)));
    }

    @Test
    public void shouldShowGamesWhenClickActionGames() {
        try {
            onView(withText(R.string.btn_ok))
                    .perform(click());
        } catch (Exception e) {
        }

        onView(withId(R.id.action_games))
                .perform(click());

        onView(allOf(isAssignableFrom(TextView.class), withParent(isAssignableFrom(Toolbar.class))))
                .check(matches(withText(R.string.title_games)));
    }

    @Test
    public void shouldShowAccountWhenClickActionAccount() {
        try {
            onView(withText(R.string.btn_ok))
                    .perform(click());
        } catch (Exception e) {
        }

        onView(withId(R.id.action_account))
                .perform(click());

        onView(allOf(isAssignableFrom(TextView.class), withParent(isAssignableFrom(Toolbar.class))))
                .check(matches(withText(R.string.title_account)));
    }

    @Test
    public void shouldShowInfoWhenClickActionInfo() {
        try {
            onView(withText(R.string.btn_ok))
                    .perform(click());
        } catch (Exception e) {
        }

        onView(withId(R.id.action_info))
                .perform(click());

        onView(withText(R.string.title_info))
                .check(matches(isDisplayed()));
    }

}

File: app/src/androidTest/java/br/com/battista/bgscore/activity/HomeActivityTest.kt
package br.com.battista.bgscore.activity

import android.Manifest
import android.content.Context
import android.content.Intent
import android.support.design.widget.BottomNavigationView
import android.support.test.filters.LargeTest
import android.support.test.rule.ActivityTestRule
import android.support.test.rule.GrantPermissionRule
import android.support.test.runner.AndroidJUnit4
import android.support.v7.widget.Toolbar
import br.com.battista.bgscore.MainApplication
import br.com.battista.bgscore.R
import br.com.battista.bgscore.helper.HomeActivityHelper
import br.com.battista.bgscore.model.User
import br.com.battista.bgscore.robot.HomeRobot
import org.hamcrest.Matchers.equalTo
import org.junit.Assert.assertNotNull
import org.junit.Assert.assertThat
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@LargeTest
@RunWith(AndroidJUnit4::class)
class HomeActivityTest {

    @Rule
    @JvmField
    var runtimePermissionRule = GrantPermissionRule.grant(Manifest.permission.READ_EXTERNAL_STORAGE)

    @Rule
    @JvmField
    var testRule = ActivityTestRule(HomeActivity::class.java, true, false)

    private lateinit var user: User
    private lateinit var context: Context
    private lateinit var homeRobot: HomeRobot

    @Before
    fun setup() {
        testRule.launchActivity(Intent())
        context = testRule.activity.applicationContext

        homeRobot = HomeRobot(context)

        val application = MainApplication.instance()

        user = HomeActivityHelper.createNewUser()
        application.clearPreferences()
        application.user = user

        homeRobot.closeWelcomeDialog()
    }

    @Test
    fun checkExistsTheElementsActivity() {
        val activity = testRule.activity

        val toolbar = activity.findViewById<Toolbar>(R.id.toolbar)
        assertNotNull(toolbar)

        val bottomNavigationView = activity.findViewById<BottomNavigationView>(R.id.bottom_navigation)
        assertNotNull(bottomNavigationView)
    }

    @Test
    fun checkExistMenuInBottomNavigationView() {
        val activity = testRule.activity

        val bottomNavigationView = activity.findViewById<BottomNavigationView>(R.id.bottom_navigation)
        assertNotNull(bottomNavigationView)

        assertThat(bottomNavigationView.menu.size(), equalTo(5))

        assertNotNull(bottomNavigationView.menu.findItem(R.id.action_home))
        assertNotNull(bottomNavigationView.menu.findItem(R.id.action_matches))
        assertNotNull(bottomNavigationView.menu.findItem(R.id.action_games))
        assertNotNull(bottomNavigationView.menu.findItem(R.id.action_more))
    }

    @Test
    fun shouldShowHomeWhenClickActionHome() {
        homeRobot.navigationToHome()
                .checkBottomBarHomeChecked()
    }

    @Test
    fun shouldShowMatchesWhenClickActionMatches() {
        homeRobot.navigationToMatches()
                .checkBottomBarMatchesChecked()
    }

    @Test
    fun shouldShowGamesWhenClickActionGames() {
        homeRobot.navigationToGames()
                .checkBottomBarGamesChecked()
    }

    @Test
    fun shouldShowAccountWhenClickActionAccount() {
        homeRobot.navigationToAccount()
                .checkBottomBarAccountChecked()
    }

}
